File: Python/sequentialTime.py
from matrixBuilders.coordinate_matrix_generator import CoordinateMatrixGenerator
from operations.original.denseMultiplication import DenseMatrixMultiplication
from matrixBuilders.transformations import MatrixTransformations
import time
from Checker.checker import Checker
import gc

N = [8, 16, 32, 64, 128, 256, 512, 1024]
density = 0.7

def test_sequential_time():
    for n in N:
        matrix = CoordinateMatrixGenerator.generate_random_coordinate_matrix(n, density)
        transform = MatrixTransformations()
        matrixA = transform.transform(matrix)

        print("Multiplying...")
        start = time.time()
        result = DenseMatrixMultiplication().multiply(matrixA, matrixA)
        end = time.time()
        elapsed_time = (end - start)
        print("N = " + str(n) + " Time = " + str(elapsed_time) + " seconds")
        Checker.test_dense(matrixA, matrixA, result)
        print("----------------------------------------------------------------------------------")
        
        del matrix
        del matrixA
        del result
        del transform
        gc.collect()

if __name__ == "__main__":
    test_sequential_time()

    
    

File: src/test/java/sequential/SequentialTime.java
package sequential;

import matrixmultiplication.Matrix;
import matrixmultiplication.MatrixBuilder;
import matrixmultiplication.checker.Checker;
import matrixmultiplication.matrixbuilder.CoordinateMatrixGenerator;
import matrixmultiplication.matrixbuilder.DenseMatrixBuilder;
import matrixmultiplication.operators.MatrixMultiplication;
import matrixmultiplication.operators.densematrixmultiplication.DenseMatrixMultiplication;

import java.util.ArrayList;
import java.util.List;

public class SequentialTime {

    public static void main(String[] args) {
        List<Integer> N = new ArrayList<>(List.of(8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096));
        MatrixMultiplication multiplier = new DenseMatrixMultiplication();

        for (Integer n: N) {

            Matrix matrixA = CoordinateMatrixGenerator.generateRandomCoordinateMatrix(n, 0.7);
            MatrixBuilder builder = new DenseMatrixBuilder(n);
            builder.setMatrix(matrixA);
            Matrix matrix = builder.get();

            long start = System.currentTimeMillis();
            Matrix result = multiplier.multiply(matrix, matrix);
            long end = System.currentTimeMillis();

            double elapsedTime = (double) (end - start);
            System.out.println("N = " + n + " Time = " + elapsedTime + " milliseconds");
            boolean check = Checker.testDense(matrix, matrix, result);
            System.out.println("The multiplication is "+ check);
            System.out.println("-----------------------------------------------------------");

            matrixA = null;
            builder = null;
            matrix = null;
            result = null;
        }
    }

}
